packages/design-system/src/components/Table/TableDataCell.jsx

```diff
@@ -0,0 +1,74 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import classNames from 'classnames';
+
+export const TableDataCell = ({
+  children,
+  className,
+  stackedTitle,
+  stackedClassName,
+  type,
+  ...tableDataProps
+}) => {
+  const classes = classNames(
+    'ds-c-table__cell',
+    className,
+    type === 'numeric' ? 'ds-c-table__cell--numeric' : null
+  );
+
+  const renderStackedTitle = () => {
+    const stackedClasses = classNames(
+      'ds-c-table--stacked__col-header',
+      'ds-u-font-weight--bold',
+      stackedClassName
+    );
+    const isValidStackedTitle = stackedTitle && stackedTitle.length > 0;
+    return (
+      { isValidStackedTitle } && (
+        <span aria-hidden="true" className={stackedClasses}>
+          {stackedTitle}
+        </span>
+      )
+    );
+  };
+
+  return (
+    <td className={classes} role="cell" {...tableDataProps}>
+      {renderStackedTitle()}
+      {children}
+    </td>
+  );
+};
+
+TableDataCell.defaultProps = {
+  type: 'text',
+};
+
+TableDataCell.propTypes = {
+  /**
+   * The table data cell contents.
+   */
+  children: PropTypes.node,
+  /**
+   * Additional classes to be added to the table cell element.
+   */
+  className: PropTypes.string,
+  /**
+   * The `headers` attribute contains a list of `id` attributes of the associated data. If there is more than one id, they are separated by spaces.
+   */
+  headers: PropTypes.string,
+  /**
+   * Additional classes to be added to the stacked Title element.
+   */
+  stackedClassName: PropTypes.string,
+  /**
+   * The stacked row title for responsive table accessiblity.
+   */
+  stackedTitle: PropTypes.string,
+  /**
+   * Type of the data, can be either text or numeric for left or right alignment respectively.
```

Comment:
Consider changing this prop from type to something like `align` or `alignment` with values of left or right. This approach is a more generic and directly applies to how the prop is used. 

Type is currently used on `TableDataCell` and `TableHeaderCell` 

Upon further consideration in our docs we recommend users to 
> Right-align numerical data in tables. You can use a text alignment utility class for this. This makes it easier to scan and compare values.

Should we remove this type prop all together and let users define classes to accomplish the same thing?

CC @bernardwang

Reply:
Good point @line47 - alignment of data can be done via utility classes. I'm good with removing the type prop.

Reply:
I would also prefer to remove

Reply:
Yeah I think we can remove the prop and make sure to include an example with right aligned numerical data

Reply:
Ah i see the default example does that already